File: solutions/java15/medium/class-and-objects-player-information-system-1/solution-approach1.java
// ──────────────────────────────────────────────────
// Link        https://www.hackerrank.com/contests/ku-uit-oops/challenges/class-and-objects-player-information-system-1/problem?isFullScreen=true
// Problem     Class and Objects - Player Information System 1
// Difficulty  Medium
// Subdomain   N/A
// Platform    HackerRank
// Language    java15
// Status      Accepted
// Submitted   2026-08-19, 08:44 a.m.
// ──────────────────────────────────────────────────

import java.io.*;
import java.util.*;

class Player{
    private String name;
    private String country;
    private String skill;
    
    Player(String name, String country, String skill){
        this.name=name;
        this.country=country;
        this.skill=skill;
    }
    void display(){
        System.out.println("Player Details:");
        System.out.println("Player Name : "+this.name);
        System.out.println("Country Name : "+this.country);
        System.out.println("Skill : "+this.skill);
    }
}
public class Solution {
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        System.out.println("Enter the player name");
        String name=sc.nextLine();
        System.out.println("Enter the country name");
        String country=sc.nextLine();
        System.out.println("Enter the skill");
        String skill=sc.nextLine();
        Player p=new Player(name, country, skill);
        p.display();
        sc.close();
    }
}

File: solutions/cpp/medium/sorting-books-by-page-count/solution-approach1.cpp
// ──────────────────────────────────────────────────
// Link        https://www.hackerrank.com/contests/ku-uit-daa/challenges/sorting-books-by-page-count/problem?isFullScreen=true
// Problem     Sorting: Sorting Books by Page Count
// Difficulty  Medium
// Subdomain   N/A
// Platform    HackerRank
// Language    cpp
// Status      Accepted
// Submitted   2026-09-17, 01:29 p.m.
// ──────────────────────────────────────────────────

#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;


int main() {
    int n;
    if (!(cin >> n)) return 0;

    int arr[n];
    for (int i = 0; i < n; i++) {
        cin >> arr[i];
    }

    for (int i = 0; i < n - 1; i++) {
        for (int j = 0; j < n - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                int temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
        }
    }

    for (int i = 0; i < n; i++) {
        cout << arr[i];
        if (i < n - 1) {
            cout << " ";
        }
    }
    cout << endl;

    return 0;
}
